### PR TITLE
Htsget POST request support

### PR DESCRIPTION
**Pull request by andersleung**
Provide classes implementing the new Htsget POST request API, as well as integrate this into the HtsgetBamFileReader class so that it makes use of the API when available.

@lbergelson 

### Things to think about before submitting:
- [x] Make sure your changes compile and new tests pass locally.
- [x] Add new tests or update existing ones:
  - A bug fix should include a test that previously would have failed and passes now.
  - New features should come with new tests that exercise and validate the new functionality.
- [ ] Extended the README / documentation, if necessary
- [x] Check your code style.
- [x] Write a clear commit title and message
  - The commit message should describe what changed and is targeted at htsjdk developers
  - Breaking changes should be mentioned in the commit message.

### CODE REVIEW

**Review by lbergelson — APPROVED** [review]
@andersleung This looks good.  I don't have any comments except for 1 question about a todo that I think might be unecessary now.

File: src/test/java/htsjdk/samtools/HtsgetBAMFileReaderTest.java [hunk 1]
```diff
@@ -199,6 +211,11 @@ public static void testQueryOverlapped(final HtsgetBAMFileReader htsgetReader) t
 
     @Test(dataProvider = "readerProvider")
     public static void testRemovesDuplicates(final HtsgetBAMFileReader htsgetReader) throws IOException {
+        // TODO: temporary workaround as reference server does not properly merge regions and remove duplicates yet
```
[the remark this answers is not in the document]
Comment (andersleung) [reply src/test/java/htsjdk/samtools/HtsgetBAMFileReaderTest.java:214]: We started the work on this but with the onramp projects (DRS and Passports) and now the file formats work there's probably not going to enough bandwidth to push it through any time soon, so it's not fixed

[the remark this answers is not in the document]
Comment (lbergelson) [reply src/test/java/htsjdk/samtools/HtsgetBAMFileReaderTest.java:214]: Is there an issue # for it we can link to somewhere to monitor the progress?  It's one of those things that's confusing in the future if we don't have somewhere to look.

[the remark this answers is not in the document]
Comment (andersleung) [reply src/test/java/htsjdk/samtools/HtsgetBAMFileReaderTest.java:214]: Created an issue and linked to it in the comment https://github.com/ga4gh/htsget-refserver/issues/27

**Comment by andersleung** [comment]
Ah sorry just force of habit. It should be ready to merge

**Comment by lbergelson** [comment]
Excellent.  I'll merge when tests pass which they definitely should because you just changed a comment..

**Pull request merged**